ui/app/mixins/cluster-route.js

```diff
@@ -96,11 +97,14 @@ export default Mixin.create({
     if (
       (!cluster.needsInit && this.routeName === INIT) ||
       (!cluster.sealed && this.routeName === UNSEAL) ||
-      (!cluster?.dr?.isSecondary && this.routeName === DR_REPLICATION_SECONDARY) ||
-      (isAuthed && this.routeName === AUTH)
+      (!cluster?.dr?.isSecondary && this.routeName === DR_REPLICATION_SECONDARY)
     ) {
       return CLUSTER;
     }
+    if (isAuthed && this.routeName === AUTH) {
```

Comment:
Nice job navigating this! This "state machine" is an unfortunately dense part of the codebase. I think this change makes sense, and I think moving the redirect elsewhere (instead of in another if here) also helps keep this more understandable.  

I had a question in another part about adding a `redirect` route - do you plan on using that more than this one case? If not, would it make sense to move that functionality to auth route and move this clause up to where `AUTH` is returned?

Reply:
One of my earlier attempts was to do it within `auth` but that is also quite dense. I could imagine a world where the redirect becomes the new URL that users link to directly, and maybe we even add some functionality that checks the namespace permissions and log you out, then redirect to auth with the `redirect_to` param intact so that the user can log in with credentials for the part of the app they're trying to get to.